### PR TITLE
browserify and caching update

### PR DESCRIPTION
**Pull request by svnlto**
this PR's in code @caolan and I have written during the last off-site. It basically caches hoodie.js but also refreshes the cache if plugin code changes. It also upgrades browserify to v8.x which comes with a slight performance improvement (~4ms).

@janl @caolan

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -46,6 +48,17 @@ exports.init = function (env_config, callback) {
         couch.start,
         installer.install,
         plugins.load,
+        function (config, callback) {
```
Comment (janl) [inline lib/index.js]: this probably wants to live in its own named function like the rest in this list?

File: test/integration/security.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ describe('block _all_dbs', function () {
   });
 
 
-  it('should log into admin', function (done) {
+  xit('should log into admin', function (done) {
```
Comment (janl) [inline test/integration/security.js]: typo?

Comment (svnlto) [inline test/integration/security.js]: nope, just a skipped test. will need to review those.

File: test/unit/helpers-pack_hoodie-test.js [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
 
   it('should be a function', function () {
     expect(hoodiejs).to.be.a(Function);
   });
 
-  it('should return a readable stream on empty cache', function (done) {
-
-    this.timeout(5000);
-
+  xit('should return a readable stream on empty cache', function (done) {
```
Comment (janl) [inline test/unit/helpers-pack_hoodie-test.js]: or maybe not a typo, not familiar with `xit`

File: lib/index.js (hunk 1 above)
Comment (svnlto) [inline lib/index.js]: yeah, forgot about that code :)

File: test/unit/helpers-pack_hoodie-test.js (hunk 3 above)
Comment (svnlto) [inline test/unit/helpers-pack_hoodie-test.js]: see above ;)

Comment (janl) [inline test/unit/helpers-pack_hoodie-test.js]: got it, thanks! :)

**Comment by janl** [comment]
Got a test fail locally after a `git clean -fdx && npm install` with `npm test`

```
  pack_hoodie
    ✓ should be a function 
    - should return a readable stream on empty cache
    - should return a cached string after first request

  Requireability
Initializing...
CouchDB started: http://127.0.0.1:6003
Waiting for CouchDB [-*----] SUCCESS 
Loading hoodie.js... 
Reading hoodie.js plugin hashes
Reading hoodie.js bundle from cache file
Error: ENOENT, no such file or directory '/Users/jan/Work/hoodie/hoodie-server/data/bundle_cache.js'
    at Object.fs.openSync (fs.js:438:18)
    at Object.fs.readFileSync (fs.js:289:15)
    at module.exports (/Users/jan/Work/hoodie/hoodie-server/lib/utils/pack_hoodie.js:131:26)
    at /Users/jan/Work/hoodie/hoodie-server/lib/index.js:54:11
    at /Users/jan/Work/hoodie/hoodie-server/node_modules/async/lib/async.js:1080:20
    at iterate (/Users/jan/Work/hoodie/hoodie-server/node_modules/async/lib/async.js:149:13)
    at /Users/jan/Work/hoodie/hoodie-server/node_modules/async/lib/async.js:160:25
    at exports.load (/Users/jan/Work/hoodie/hoodie-server/lib/core/plugins.js:22:3)
    at /Users/jan/Work/hoodie/hoodie-server/node_modules/async/lib/async.js:1080:20
    at iterate (/Users/jan/Work/hoodie/hoodie-server/node_modules/async/lib/async.js:149:13)
```

**Comment by svnlto** [comment]
can you retry with removing the `/data` directory before.

**Comment by janl** [comment]
That fixed it, yay!

**Comment by svnlto** [comment]
but it broke something else. please revert!

**Comment by janl** [comment]
may bad, sorry, reverting.

**Pull request merged**